test/unit/vl-map-wms-layer.test.html

```diff
@@ -0,0 +1,80 @@
+<!doctype html>
+<html>
+
+<head>
+  <meta charset="utf-8">
+  <script src="../../../@webcomponents/webcomponentsjs/webcomponents-lite.js"></script>
+  <script src="../../../web-component-tester/browser.js"></script>
+
+  <script type="module" src="../../src/vl-map-all.js"></script>
+  <script src="/node_modules/sinon/pkg/sinon.js"></script>
+</head>
+
+<body>
+  <test-fixture id="vl-map-fixture">
+    <template>
+      <vl-map></vl-map>
+    </template>
+  </test-fixture>
+
+  <test-fixture id="vl-map-wms-layer-fixture">
+    <template>
+      <vl-map>
+        <vl-map-wms-layer></vl-map-wms-layer>
+      </vl-map>
+    </template>
+  </test-fixture>
+
+  <test-fixture id="vl-map-wms-layer-hidden-fixture">
+    <template>
+      <vl-map>
+        <vl-map-wms-layer data-vl-hidden></vl-map-wms-layer>
+      </vl-map>
+    </template>
+  </test-fixture>
+
+  <script type="module">
+    suite('vl-map-wms-layer', () => {
+      test('de kaartlaag zal pas angemaakt worden na constructie zodat op moment van constructie nog niet al de attributen gekend moeten zijn', (done) => {
+        const types = ['image', 'tiled'];
+        types.forEach((type, index) => {
+          const map = fixture('vl-map-fixture');
+          const layer = document.createElement('vl-map-image-wms-layer');
+          layer.setAttribute('data-vl-url', 'http://dummy/wms-adjusted');
+          layer.setAttribute('data-vl-layers', 'layer1');
+          layer.setAttribute('data-vl-styles', 'style1,style2');
+          layer.setAttribute('data-vl-version', '1.1.1');
+          layer.setAttribute('data-vl-opacity', '0.75');
+          layer.setAttribute('data-vl-min-resolution', '10');
+          layer.setAttribute('data-vl-max-resolution', '1000');
+          layer.setAttribute('data-vl-name', 'adjusted');
+          assert.isUndefined(layer.source);
+          assert.isUndefined(layer.layer);
+          map.appendChild(layer);
+          setTimeout(() => {
+            assert.isDefined(layer.source);
+            assert.isDefined(layer.layer);
+
+            if (types.length == index+1) {
+              done();
+            }
+          });
+        });
+      });
+
+      // TODO faalt
```

Comment:
Dit moet ook nog opgelost worden, waarschijnlijk lukt dit niet omdat een wms-layer geen component is.

Reply:
De testen zijn al gepushed maar waren nog niet afgewerkt.